### PR TITLE
Delimiter selector updated

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
* Delimiter selector made more robust and usable for all languages
* Added a lot of language specific play/test files. They have very little in them right now but the purpose is that each language should have a file with all the common structures making it easier to test while developing.

### CODE REVIEW

File: src/languages/cpp.ts [hunk 1]
```diff
@@ -1,109 +1,112 @@
-import {
```
Comment (pokey) [inline src/languages/cpp.ts:1]: Not sure what happened with this diff. Did you change line endings or something? Same with all the other language parse tree pattern files files

**Review by pokey — COMMENTED** [review]
I like this idea. A few things

- [x] Looks like the line endings got messed up or something? Strange diffs
- [ ] Looks like a few junk files appeared, eg `tmp.tsx` and a couple empty files. The markdown one also looks strange
- [ ] Please add tests for the things you fixed
- [ ] I'd be tempted to put all the files, including hat-test, into a `data/playground` directory. I think "test" is probably a confusing name because they're not actually tests. Prob my fault for calling it "hat-test". I think I'd maybe go `data/playground/hats.txt`, `data/playground/typescript.ts`, etc?

File: src/languages/cpp.ts (hunk 1 above)
Comment (AndreasArvidsson) [reply src/languages/cpp.ts:1]: I'm trying to switch to LF and something rent wrong.

**Comment by AndreasArvidsson** [comment]
Now I think everything is fixed :)

File: src/test/suite/fixtures/recorded/languages/java/chuckTypeNear.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    - anchor: {line: 3, character: 1}
+      active: {line: 3, character: 1}
+  marks:
+    default.n:
+      start: {line: 1, character: 24}
+      end: {line: 1, character: 28}
+finalState:
+  documentContents: |-
+    public class Java {
+        private Java( name) {
```
Comment (pokey) [inline src/test/suite/fixtures/recorded/languages/java/chuckTypeNear.yml:27]: This one is a bit annoying but probably out of scope for this PR

File: src/util/nodeSelectors.ts [hunk 3]
```diff
@@ -75,47 +75,63 @@ export function unwrapSelectionExtractor(
...
-      : null;
-
-  return {
-    ...simpleSelectionExtractor(editor, node),
-    context: {
-      trailingDelimiterRange,
-    },
+        );
+      } else {
+        trailingDelimiterRange = makeRangeFromPositions(
```
Comment (pokey) [inline src/util/nodeSelectors.ts:123]: What if we walked up parents until we find one with a previous sibling and then ask for end of previous sibling there? Not sure it's worth the effort but might make white space a bit cleaner

**Review by pokey — APPROVED** [review]
This is really nice.  Left a couple minor suggestions

File: src/languages/csharp.ts [hunk 4]
```diff
@@ -213,33 +205,27 @@ const getMapMatchers = {
...
     makeDelimitedSelector("(", ")")
   ),
-  namedFunction: typeMatcher(...NAMED_FUNCTION_TYPES),
-  functionName: composedMatcher([
-    typedNodeFinder(...NAMED_FUNCTION_TYPES),
-    getNameNode,
-  ]),
-  comment: matcher(typedNodeFinder("comment")),
-  regularExpression: matcher(typedNodeFinder("regex")),
-  type: matcher(findTypeNode, selectWithLeadingDelimiter),
```
[the remark this answers is not in the document]
Comment (AndreasArvidsson) [reply src/languages/csharp.ts:241]: c# don't have any delimiter for their type. Just like java it's just a space. One could argue that we could include the space as a delimiter but we don't do that for java or c++ so I just wanted to be consistent. And that delimiter is trailing not leading so the previous implementation makes no sense.

File: src/test/suite/fixtures/recorded/languages/java/chuckTypeNear.yml (hunk 2 above)
Comment (AndreasArvidsson) [reply src/test/suite/fixtures/recorded/languages/java/chuckTypeNear.yml:27]: One way we could avoid that would be to include a trailing space as a delimiter. But I don't think spaces are tokens so we going to have to do that text wise without the tree sitter.

File: src/util/nodeSelectors.ts (hunk 3 above)
Comment (AndreasArvidsson) [reply src/util/nodeSelectors.ts:123]: We can't try that

File: src/test/suite/fixtures/recorded/languages/java/chuckTypeNear.yml (hunk 2 above)
Comment (pokey) [reply src/test/suite/fixtures/recorded/languages/java/chuckTypeNear.yml:27]: Yeah I think in the future we could find a way to let token spacing treatment handle this one. Happy to call this out of scope for this PR

File: src/util/nodeSelectors.ts (hunk 3 above)
Comment (pokey) [reply src/util/nodeSelectors.ts:123]: Did you mean to say "we _can_ try that?"

Comment (AndreasArvidsson) [reply src/util/nodeSelectors.ts:136]: I don't think we have to. So far no language has need of this with the current implementation.

File: src/test/suite/fixtures/recorded/languages/java/chuckTypeNear.yml (hunk 2 above)
Comment (AndreasArvidsson) [reply src/test/suite/fixtures/recorded/languages/java/chuckTypeNear.yml]: This problem is now fixed :)

File: src/languages/csharp.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (AndreasArvidsson) [reply src/languages/csharp.ts:241]: New selector removes but white space and also used for c#

**Pull request merged**